File: include/apl/errors/ValueError.java
package APL.errors;

import APL.types.*;

public class ValueError extends APLError {
  public ValueError(std::string s) {
    super(s);
  }
  public ValueError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public ValueError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }
}

File: APL/errors/ValueError.cpp
#include <APL/ValueError.h>

namespace APL::errors
{
    // using namespace APL::types;

    ValueError::ValueError(std::string const& s) : APLError(s)
    {
    }

    ValueError::ValueError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    ValueError::ValueError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }
}
